var/spack/repos/builtin/packages/llvm-amdgpu/package.py

```diff
@@ -228,8 +230,8 @@ def cmake_args(self):
             args.append("-DLLVM_LINK_LLVM_DYLIB:Bool=ON")
 
         # Get the GCC prefix for LLVM.
-        if self.compiler.name == "gcc":
-            compiler = Executable(self.compiler.cc)
+        if self.pkg.compiler.name == "gcc":
```

Comment:
I tried to intercept missing an attribute in a `Package` when using the adapter, to emit a better error message in 46b940cea4fe10f649a8e9a597ab7c1b1bcfd86b - which is the opposite direction as what is discussed here. Since `Builder` is a new class introduced in this PR, and not something existing that we moved, I would avoid placeholder attributes that need to be kept in sync with `Package`.